### PR TITLE
Final changes to support WAB in non-SSO mode

### PR DESCRIPTION
**Pull request by ginach**

### CODE REVIEW

File: src/OneDriveSdk.WinStore/Authentication/WebAuthenticationBrokerWebAuthenticationUi.cs [hunk 1]
```diff
@@ -41,7 +41,15 @@ public class WebAuthenticationBrokerWebAuthenticationUi : IWebAuthenticationUi
...
 
             // Attempt to authentication without prompting the user first.
-            result = await this.AuthenticateAsync(requestUri, callbackUri, WebAuthenticationOptions.SilentMode);
+            try
+            {
+                result = await this.AuthenticateAsync(requestUri, callbackUri, WebAuthenticationOptions.SilentMode);
+            }
+            catch (Exception)
+            {
+                // WebAuthenticationBroker can throw an exception in silent authentication mode when not using SSO and
```
Comment (peternied) [inline src/OneDriveSdk.WinStore/Authentication/WebAuthenticationBrokerWebAuthenticationUi.cs]: Could these exception still be logged?

File: src/OneDriveSdk.WinStore/Extensions/OneDriveClientExtensions.cs [hunk 2]
```diff
@@ -40,13 +40,13 @@ public static class OneDriveClientExtensions
             string[] scopes,
             IHttpProvider httpProvider = null)
         {
-            return OneDriveClientExtensions.GetClientUsingWebAuthenticationBroker(appId, scopes, null, httpProvider);
+            return OneDriveClientExtensions.GetClientUsingWebAuthenticationBroker(appId, null, scopes, httpProvider);
```
Comment (peternied) [inline src/OneDriveSdk.WinStore/Extensions/OneDriveClientExtensions.cs]: Nit: missing label on this null

**Comment by aclev** [comment]
:+1:

**Comment by peternied** [comment]
:+1:

File: src/OneDriveSdk.WinStore/Extensions/OneDriveClientExtensions.cs (hunk 2 above)
Comment (ginach) [inline src/OneDriveSdk.WinStore/Extensions/OneDriveClientExtensions.cs]: Have a fix in the next PR I'm about to send out.

**Pull request merged**